### PR TITLE
Add option to disable cursor rotation.

### PR DESCRIPTION
**Pull request by alula**
This is so annoying, so I've added option to toggle it.

### CODE REVIEW

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 1]
```diff
@@ -143,6 +149,10 @@ private void load(OsuConfigManager config, TextureStore textures, OsuColour colo
                 cursorScale = config.GetBindable<double>(OsuSetting.MenuCursorSize);
                 cursorScale.ValueChanged += newScale => cursorContainer.Scale = new Vector2((float)newScale * base_scale);
                 cursorScale.TriggerChange();
+
+                cursorRotate = config.GetBindable<bool> (OsuSetting.CursorRotation);
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: why are you doing this in `Cursor` and not `MenuCursor`?

Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: i have no idea how to get OsuConfigManager instance...

Comment (jorolf) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: with dependency injection
(look at other examples)

Comment (jorolf) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: you used it in `DetailSettings` 🤔

Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: yeah, noticed and moved, now testing...

Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: isn't that a detail?

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 2]
```diff
@@ -20,13 +20,16 @@ public class MenuCursor : CursorContainer
...
+        private bool dragRotating;
+
         private bool dragging;
 
         private bool startRotation;
 
         protected override bool OnMouseMove(InputState state)
         {
-            if (((Cursor)ActiveCursor).DragRotating) {
+            if (dragRotating) {
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: just use `cursorRotate`

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 3]
```diff
@@ -20,32 +20,37 @@ public class MenuCursor : CursorContainer
...
-                    float degrees = (float)MathHelper.RadiansToDegrees(Math.Atan2(-offset.X, offset.Y)) + 24.3f;
-
-                    // Always rotate in the direction of least distance
-                    float diff = (degrees - ActiveCursor.Rotation) % 360;
-                    if (diff < -180) diff += 360;
-                    if (diff > 180) diff -= 360;
-                    degrees = ActiveCursor.Rotation + diff;
-
-                    ActiveCursor.RotateTo(degrees, 600, Easing.OutQuint);
+            if (dragRotating) {
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: and put `{` on new lines

Comment (ColdVolcano) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: Put the brackets in a new line to match the standards

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 4]
```diff
@@ -102,10 +107,19 @@ protected override void PopOut()
             ActiveCursor.ScaleTo(0, 500, Easing.In);
         }
 
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config, TextureStore textures, OsuColour colour)
```
Comment (ColdVolcano) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: You only use `OsuConfigManager`, remove the other arguments

File: osu.Game/Overlays/Settings/Sections/Graphics/DetailSettings.cs [hunk 5]
```diff
@@ -1,10 +1,26 @@
 ﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+using osu.Framework.Allocation;
+using osu.Game.Configuration;
+
```
Comment (ColdVolcano) [inline osu.Game/Overlays/Settings/Sections/Graphics/DetailSettings.cs]: Extra new line

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 6]
```diff
@@ -20,32 +20,37 @@ public class MenuCursor : CursorContainer
...
-
-                    // Always rotate in the direction of least distance
-                    float diff = (degrees - ActiveCursor.Rotation) % 360;
-                    if (diff < -180) diff += 360;
-                    if (diff > 180) diff -= 360;
-                    degrees = ActiveCursor.Rotation + diff;
-
-                    ActiveCursor.RotateTo(degrees, 600, Easing.OutQuint);
+            if (dragRotating) {
+                if (dragging) {
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: combine these if statements..?

Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: did before ;v

File: osu.Game/Graphics/Cursor/MenuCursor.cs (hunk 3 above)
Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: i'm java programmer and i hate allman...
but code style is code style ;v

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 7]
```diff
@@ -102,10 +105,18 @@ protected override void PopOut()
             ActiveCursor.ScaleTo(0, 500, Easing.In);
         }
 
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
+        {
+            cursorRotate = config.GetBindable<bool> (OsuSetting.CursorRotation);
+            cursorRotate.TriggerChange();
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: that is also redundant

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 8]
```diff
@@ -20,32 +20,35 @@ public class MenuCursor : CursorContainer
...
+                Debug.Assert (state.Mouse.PositionMouseDown != null);
 
                 // don't start rotating until we're moved a minimum distance away from the mouse down location,
                 // else it can have an annoying effect.
-                startRotation |= Vector2Extensions.Distance(state.Mouse.Position, state.Mouse.PositionMouseDown.Value) > 30;
+                startRotation |= Vector2Extensions.Distance (state.Mouse.Position, state.Mouse.PositionMouseDown.Value) > 30;
 
-                if (startRotation)
-                {
+                if (startRotation) {
```
Comment (ColdVolcano) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: this bracket is still here :/

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 9]
```diff
@@ -20,32 +20,35 @@ public class MenuCursor : CursorContainer
...
-                    if (diff < -180) diff += 360;
-                    if (diff > 180) diff -= 360;
+                    if (diff < -180)
+                        diff += 360;
+                    if (diff > 180)
+                        diff -= 360;
                     degrees = ActiveCursor.Rotation + diff;
 
-                    ActiveCursor.RotateTo(degrees, 600, Easing.OutQuint);
+                    ActiveCursor.RotateTo (degrees, 600, Easing.OutQuint);
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: random space?

Comment (jorolf) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: everywhere

Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: MonoDevelop/Xamarin Studio inserts it ;v

Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: Changed code style in settings, should be ok now...

**Comment by gabixdev** [comment]
And?

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 10]
```diff
@@ -102,10 +106,17 @@ protected override void PopOut()
...
+        private void load(OsuConfigManager config)
+        {
+            cursorRotate = config.GetBindable<bool>(OsuSetting.CursorRotation);
+        }
+
         public class Cursor : Container
         {
             private Container cursorContainer;
             private Bindable<double> cursorScale;
+
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: why did you add this?

File: osu.Game/Graphics/Cursor/MenuCursor.cs [hunk 11]
```diff
@@ -41,8 +43,10 @@ protected override bool OnMouseMove(InputState state)
 
                     // Always rotate in the direction of least distance
                     float diff = (degrees - ActiveCursor.Rotation) % 360;
-                    if (diff < -180) diff += 360;
-                    if (diff > 180) diff -= 360;
+                    if (diff < -180)
```
Comment (jorolf) [inline osu.Game/Graphics/Cursor/MenuCursor.cs]: and change this?

**Review by jorolf — CHANGES_REQUESTED** [review]
other than that, _and the fact that you changed the csproj file_, it's ok

File: osu.Game/Graphics/Cursor/MenuCursor.cs (hunk 10 above)
Comment (gabixdev) [reply osu.Game/Graphics/Cursor/MenuCursor.cs]: formatter...

**Review by peppy — CHANGES_REQUESTED** [review]
Please fix the remaining code formatting issues

**Comment by peppy** [comment]
@gabixdev still alive? this one could be merged with the simple fixes

**Comment by gabixdev** [comment]
yes

**Review by peppy — APPROVED** [review]
